templates/kotlin/src/main/kotlin/io/appwrite/Permission.kt.twig

```diff
@@ -1,21 +1,61 @@
 package {{ sdk.namespace | caseDot }}
 
-class Permission {
-    companion object {
-        fun read(role: String): String {
-            return "read(\"${role}\")"
-        }        
-        fun write(role: String): String {
-            return "write(\"${role}\")"
-        }        
-        fun create(role: String): String {
-            return "create(\"${role}\")"
-        }        
-        fun update(role: String): String {
-            return "update(\"${role}\")"
-        }        
-        fun delete(role: String): String {
-            return "delete(\"${role}\")"
-        }        
+/**
+ * Helper class for constructing permissions
+ */
+object Permission {
+    /**
+     * Constructs a permission, that allows [role] to read X
```

Comment:
Instead of X, let's specify, for all the permissions methods:

```suggestion
     * Constructs a permission, that allows [role] to read a specific resource
```

Reply:
Agree, but then let's just make it "a resource"